File: DSA/Assignment_6/stack_ll.c
#include <stdio.h>
#include <stdlib.h>
typedef struct Node
{
    int data;
    struct Node *next;
}node;

node *createNode(int data)
{
    node *newNode=(node *)malloc(sizeof(node));
    newNode->data=data;
    newNode->next=NULL;
    return newNode;
}

void insert_ele(node* *head, int data)
{
    node *newNode=createNode(data);
    if(*head==NULL)
    {
        *head=newNode;
        return;
    }
    newNode->next=*head;
    *head=newNode;
}

void del_head(node* *head)
{
    node *temp=*head;
    *head=(*head)->next;
    free(temp);
}

//Stack implementation
int isEmpty(node* *stack)
{
    return *stack==NULL;
}

int isFull(node* *stack, int max_size)


void push(node* *stack, int data)
{
    insert_ele(stack, data);
}

int pop(node* *stack)
{
    if(isEmpty(stack))
    {
        printf("Stack underflow!");
        return -1;
    }
    del_head(stack);
}

int peek(node* *stack)
{
    if(isEmpty(stack))
    {
        printf("Stack underflow!");
        return -1;
    }
    return (*stack)->data;
}

void print_stack(node* *stack)
{
    if(isEmpty(stack))
    {
        printf("Stack underflow!");
        return;
    }
    node *temp=*stack;
    while(temp)
    {
        printf("%d\n",temp->data);
        temp=temp->next;
    }
}

int main()
{
    node *stack=NULL;
    int n, ele;
    printf("Enter size of stack:");
    scanf("%d",&n);
    for(int i=0;i<n;i++)
    {
        printf("Enter element:");
        scanf("%d",&ele);
        push(&stack, ele);
    }
    printf("Original Stack:\n");
    print_stack(&stack);

    //pop top element of stack
    printf("Popped top element:%d\n",pop(&stack));

    //peek top element of stack
    printf("Top element of stack:%d\n",peek(&stack));

    //print whole stack
    printf("Modified Stack:\n");
    print_stack(&stack);
    
    return 0;
}

File: DSA/Assignment_6/stack_ll.py
class Node:
    def __init__(self, data):
        self.data=data
        self.next=None

class stackLinkedList:
    def __init__(self):
        self.head=None

    @staticmethod
    def createNode(self, data):
        return Node(data)
    
    def insert_ele(self, data):
        newNode=self.createNode(data)
        if self.head is None:
            self.head=newNode
            return
        newNode.next=self.head
        self.head=newNode

    def del_head(self):
        if self.head is None:
            return None
        temp=self.head
        self.head=self.head.next
        return temp.data
    
    def isEmpty(self):
        return self.head is None
    
    def isFull(self):
        i